nemo/collections/asr/data/ssl_dataset.py

```diff
@@ -0,0 +1,723 @@
+        return None
+
+    def __init__(
+        self,
+        noise_manifest: str | None = None,
+        batch_augmentor: Any | None = None,
+        max_audio_cache: int = 5,
+        min_audio_len_secs: float = 1.0,
+        pad_audio_mode: str = 'repeat',
+        **kwargs,
+    ):
+        super().__init__(bos_id=0, manifest_parse_func=_parse_manifest_item, **kwargs)
+        self.noise_manifest = noise_manifest
+        self.batch_augmentor = batch_augmentor
+        self.noise_data = load_noise_manifest(noise_manifest)
+        self._audio_cache = OrderedDict()
+        self._max_audio_cache = max_audio_cache
+        self.min_audio_len_secs = min_audio_len_secs
+        self.pad_audio_mode = pad_audio_mode
+
+    def _build_sample(self, tup):
+        """Builds the training sample by combining the data from the WebDataset with the manifest info."""
+        audio_bytes, audio_filename, offset_id = tup
+
+        # Grab manifest entry from self.manifest_preprocessor.collection
+        file_id, _ = os.path.splitext(os.path.basename(audio_filename))
+        manifest_idx = self.manifest_processor.collection.mapping[file_id][offset_id]
+        manifest_entry = self.manifest_processor.collection[manifest_idx]
+
+        offset = manifest_entry.offset
+        if offset is None:
+            offset = 0
+
+        failed = False
+        try:
+            # Convert audio bytes to IO stream for processing (for SoundFile to read)
+            audio_filestream = io.BytesIO(audio_bytes)
+            audio = self.featurizer.process(
+                audio_filestream,
+                offset=offset,
+                duration=manifest_entry.duration,
+                trim=self.trim,
+                orig_sr=manifest_entry.orig_sr,
+            )
+            audio_filestream.close()
+        except Exception as e:
+            logging.warning(f"Error reading audio sample: {manifest_entry}, with exception: {e}.")
+            failed = True
+
+        if audio.size(0) == 0:
+            logging.warning(f"Loaded audio has zero length: {manifest_entry}.")
+            failed = True
+        elif audio.isnan().any():
+            logging.warning(f"Loaded audio has NaN values: {manifest_entry}.")
+            failed = True
+        elif audio.abs().max() == 0:
+            logging.warning(f"Loaded audio has all zero values: {manifest_entry}.")
+            failed = True
+
+        if failed:
+            if len(self._audio_cache) > 0:
+                idx, audio = self._audio_cache.popitem(last=False)
+                logging.warning(
+                    f"Error reading audio sample: {manifest_entry}, returning audio from cache for sample: {self.manifest_processor.collection[idx]}"
+                )
+            else:
+                logging.warning(f"Error reading audio sample: {manifest_entry}, returning dummy audio for sample.")
+                audio = 0.1 * torch.ones(self.featurizer.sample_rate, dtype=torch.float32)
```

Comment:
I dont think we should be doing this - it can silently derail model training

Reply:
Agreed, it was added to use the SSL data that wan't completely cleaned up. I just removed them.